Address compiler warnings

yokattana opened the pull request:
(Compiling with GCC 4.8.4 on LXSS/Ubuntu)

 - unused results
 - incorrect size argument in fgetws

One unused result warning, for dup(), has not been addressed I'm not familiar with its use.

andmarti1424 commented:
Shouldnt "exit_app" function be called, to free malloc'ed structures and end ncurses correctly?

yokattana commented:
Good catch, I wasn't aware of that function. I haven't used ncurses before so I don't know about these things yet.

Addressed in new commit. Also switched to `sc_error` for the messages, as I suppose going through the existing systems is better than writing directly to standard error.

andmarti1424 commented:
You are making such a great job!!
